documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,127 @@
+# Repository Guidelines for Cadl Specs
+
+## Purpose
+
+We need to formulate a strategy for checking in actual Cadl service specs for Azure. Service teams need to know where
+and how to check these in and tooling needs to know how to find and consume them. While OpenAPI specs traditionally have been just spec
+documents, Cadl "specs" could include service-specific functions, decorators and so forth that elevate them to the level of a
+library, and to maximize the benefits of Cadl, this approach needs to be supported as a first-class citizen.
+
+## Repository
+
+Cadl can co-exist with Swagger within the existing `azure-rest-api-specs` repository. This approach will make it easier to generate Swagger artifacts without needing to sync between repos at the downside of having to live with any baggage associated with the repo as it ages. There are currently over 1,000 issues and 500 open PRs in this repo.
+
+## Structure Overview
+
+This proposal strives to align with [Azure SDK Repo Structure Guidelines](https://azure.github.io/azure-sdk/policies_repostructure.html).
+
+We eliminate the distinction between data-plane and resource-manager, treating all packages as siblings under the service family folder. This is what the SDKs do today. The legacy `data-plane` and `resource-manager` folders where Swagger specs live will remain.
+
+A given Cadl folder could represent any of the following scenarios:
+
+- an SDK and service spec (simplest, common for small services)
+- a service spec only
+- an SDK spec that aggregates many small service specs (one SDK to many microservices)
+- an SDK spec that extracts a portion of a large service spec (many SDKs to a single large service)
```

Comment:
Can you give some examples of when these would be useful?

I wonder if we want Cadl docs in the azure-rest-api-specs repo that don't describe an Azure REST API.

Reply:
@tjprescott would you respond Mike's questions? I have the same question.  What do you mean 'SDK spec'?

Reply:
@mikekistler an example might be where a large service spec contains service A and B, but they want an SDK for A and an SDK for B. They could simply extract the SDK shape for A and B from the service spec using the `@client` decorators (and other DPG decorators). Alternatively, they could simply model A and B as separate services and use the first option (service spec and SDK definition are one and the same). 

The mention here isn't to say that this is a common (or recommended) practice, merely that it's a possibility (and is compatible with this proposal).

Reply:
An SDK spec means that the service definition lives elsewhere and that the given cadl describes only the shape of an SDK.